docs/users.rst

```diff
@@ -1408,12 +1408,40 @@ Before OSBS can pin your pullspecs, it first needs to find them. Because
 it is practically impossible to tell if a string is a pullspec, atomic-reactor
 has a predefined set of locations where it will look for pullspecs.
 
-*Locations shown as jq queries.*
+1. metadata.annotations.containerImage anywhere in the file
 
-1. ``.metadata.annotations.containerImage``
-2. ``.spec.install.spec.deployments[].spec.template.spec.containers[]``
-3. ``.spec.install.spec.deployments[].spec.template.spec.initContainers[]``
-4. ``.env[] | select(.name | test("RELATED_IMAGE_"))`` for each of [2], [3]
+   jq: ``.. | .metadata?.annotations.containerImage | select(. != null)``
+
+2. All containers in each deployment
+
+   jq: ``.spec.install.spec.deployments[].spec.template.spec.containers[]``
+
+3. All initContainers in each deployment
+
+   jq: ``.spec.install.spec.deployments[].spec.template.spec.initContainers[]``
+
+4. All RELATED\_IMAGE\_* variables for all containers and initContainers
+
+   jq: ``.env[] | select(.name | test("RELATED_IMAGE_"))`` for each of [2], [3]
+
+5. OSBS tries to do the (almost) impossible and find all pullspecs in all
+   annotations. See *heuristic annotations* below.
+
+**Heuristic annotations**
+
+OSBS will attempt to extract all pullspecs from all attributes of each
+``metadata.annotations`` object in a CSV. If an attribute contains more than
+one pullspec (as text, e.g. comma-separated), all of them should be found.
+How OSBS does that is not important for the purposes of this document, but an
```

Comment:
Here we should either point to the code performing the heuristics (with a PR or commit reference) or mention the "how" for the heuristic.